chef/cookbooks/nova/recipes/config.rb

```diff
@@ -107,40 +107,28 @@
 
 keystone_settings = KeystoneHelper.keystone_settings(node, @cookbook_name)
 
-ceph_user = node[:nova][:rbd][:user]
-ceph_uuid = node[:nova][:rbd][:secret_uuid]
-
 cinder_servers = search_env_filtered(:node, "roles:cinder-controller") || []
 if cinder_servers.length > 0
   cinder_server = cinder_servers[0]
   cinder_insecure = cinder_server[:cinder][:api][:protocol] == 'https' && cinder_server[:cinder][:ssl][:insecure]
 
   if node.roles.include? "nova-multi-compute-kvm"
-    include_ceph_recipe = false
-
+    rbd_enabled = false
     cinder_server[:cinder][:volumes].each do |volume|
-      next if volume['backend_driver'] != "rbd"
-
-      # if include_ceph_recipe is already true, avoid re-entering the if (and executing a slow search)
-      if volume['rbd']['use_crowbar'] && !include_ceph_recipe
-        ceph_env_filter = " AND ceph_config_environment:ceph-config-default"
-        ceph_servers = search(:node, "roles:ceph-osd#{ceph_env_filter}") || []
-        if ceph_servers.length > 0
-          include_ceph_recipe = true
-        end
-      end
-    end
-
-    if include_ceph_recipe
-      include_recipe('ceph::nova')
-      ceph_user = node['ceph']['nova-user']
-      ceph_uuid = node['ceph']['nova-uuid']
+      rbd_enabled = true if volume['backend_driver'] == "rbd"
     end
   end
 else
   cinder_insecure = false
 end
 
+if rbd_enabled
+  include_recipe "nova::ceph"
+end
+
+ceph_user = node[:nova][:rbd][:user]
+ceph_uuid = node[:nova][:rbd][:secret_uuid]
```

Comment:
Shouldn't we drop the uuid bit with the work happening in https://github.com/crowbar/barclamp-cinder/pull/249 ? As I mentioned in https://github.com/crowbar/barclamp-nova/pull/449#discussion_r28977056 I believe that nova will always use this UUID instead of the one provided by cinder, and so we shouldn't define any UUID in nova.conf.